packages/headless/src/features/search/search-actions-thunk-processor.ts

```diff
@@ -172,29 +173,29 @@ export class AsyncSearchThunkProcessor<RejectionType> {
     }
 
     const {enableDidYouMean, automaticallyCorrectQuery} = state.didYouMean;
-    const {results, queryCorrections} = successResponse;
+    const {results, queryCorrections, changedQuery} = successResponse;
 
-    const shouldBeAutomaticallyRetried =
+    const shouldExecuteClassicDidYouMeanAutoCorrection =
+      enableDidYouMean === true &&
+      automaticallyCorrectQuery === true &&
       results.length === 0 &&
-      queryCorrections.length !== 0 &&
-      enableDidYouMean &&
```

Comment:
Why do we check for `enableDidYouMean` here? If someone has `enableDidYouMean` set in their query param overrides, does it mean it wouldn't auto correct here?

Reply:
If there's a query override for enableDidYouMean, we will probably have a bug, yes. 

That's a general problem with some query param which the UI * must * know about to do it's job properly, but that people can just decide to override in the backend without the front end knowing about it.

There's a couple of those query parameter actually (pagination comes to mind) that we know about, but the problems don't come too often...